### ISSUE TITLE
LGPL License - Laymans terms?

### ISSUE THREAD

**Issue by togosh** [rexxar-tc/SEServerExtender#8]
LGPL v3
https://github.com/rexxar-tc/SEServerExtender/blob/master/LICENSE
added to SEServerExtender by @fvgoulet Jun 8, 2014

---

References/Resources:

GNU Lesser General Public License
Wiki - https://en.wikipedia.org/wiki/GNU_Lesser_General_Public_License  

LGPL v2.1
https://tldrlegal.com/license/gnu-lesser-general-public-license-v2.1-(lgpl-2.1)#summary

LGPL 2.1 vs LGPL 3.0 Advantages and Disadvantages
http://programmers.stackexchange.com/questions/189668/lgpl-2-1-vs-lgpl-3-0-advantages-and-disadvantages

LGPL and Java
http://www.gnu.org/licenses/lgpl-java.html

---
1. Im not a lawyer, in laymans terms, what does the LGPL v3 license mean for?:  
   A. SEServerExtender in General  
   B. Forks of SEServerExtender  
   C. Plugins for SEServerExtender  
   D. Dedicated Servers that use a combination of A, B, C above

**Comment by rexxar-tc** [comment]
Under LGPL, you can use, modify, distribute, or sell the code under the conditions:
- The original license must be maintained, you cannot change form LGPL to GNU, for instance
- You must include the full license text
- You cannot claim the entire codebase as your own, you should link back to wherever you branched from
- You must maintain change logs (github commits are enough)
- Make a good faith effort to contact the author

In this case, you can branch SESE or Extender, and maintain your own releases. Since you're required to also use LGPL, that would mean I could branch your repo, or pull code if you've done something interesting (I don't do this without asking, of course). You can run your own version of SESE and Essentials, and as long as you link back to me or the SEServerExtender account.

Creating your own plugin is a different story. If you don't compile any SESE resources into your plugin or redistribute the DLLs, you're free to do whatever you want. Simply using the SESE API doesn't fall under any licensing (I think); you aren't bound to LGPL. If you do include LGPL code, you're required to use the LGPL license and publish your source. For example, my NoGrief plugin uses code from Essentials as a framework; I have to publish it under LGPL.

Basically it all boils down to "don't be a dick". Don't claim the code is yours, and let me know if you're selling your branch. Beyond that, have fun.

SESE is a community project; the license is to protect us if someone tries to do something like steal the whole project. No one will care if you fudge the rules and use some SESE code in your closed-source plugin. If in doubt, just ask.

Read this: https://tldrlegal.com/license/gnu-lesser-general-public-license-v2.1-%28lgpl-2.1%29

On another note, don't link back to me as the source, use the SEServerExtender account. That's the official repo, and dodexahedron said he'd give me push access once the holidays are over. Eventually my code will go there along with everyone else that contributes.

**Comment by togosh** [comment]
@rexxar-tc++ 
Thank you for explaining all of this!

On the topic of Essentials, I dont think I saw a license file?

**Comment by rexxar-tc** [comment]
Huh, it doesn't have a license file. Well, just assume it's LGPL, I'll get around to it eventually.
